src/pybind/mgr/restful/api/request.py

```diff
@@ -77,6 +77,11 @@ def post(self, **kwargs):
         """
         Pass through method to create any request
         """
+        if isinstance(request.json, list):
+            command_list = []
+            for command in request.json:
+                command_list.append(command)
+            return context.instance.submit_request([command_list], **kwargs)
```

Comment:
This will just allow you to run multiple commands in parallel, not sequentially. Is that what you wanted? We should pass the full list structure directly to the submit_request function instead; that would give you the ability to execute multiple commands in parallel as well as sequentially.

The format that `submit_request` accepts is documented, here:

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/restful/module.py#L37

i.e. when passing the full structure, the endpoint would accept a list of lists where the top-level list defines sequential execution of commands and the lists inside the top-level list would define parallel execution of commands.

We may want to error out if the provided list is not a list of lists.

Reply:
Hi Boris,
Thank you advice, I has new commit that can handles parallel as well as sequential execution of
commands. And It's also can support old version format to call request api.